bin/shell/osh.pl

```diff
@@ -1130,19 +1130,59 @@ sub main_exit {
 # now build the real command
 my @command;
 
+# if we are doing a password login, find the password(s)
+
+# for autologin, "-1" means "try the main password, then try the fallbacks", 0 means "try only the main password", and N means "try only the Nth fallback password"
+my $forcePasswordId = -1;
+
+if ($userPasswordClue) {
+
+    # locate main password file
+    my $fnretpass = OVH::Bastion::get_passfile(hint => $userPasswordClue, context => $userPasswordContext, self => ($remoteself || $sysself), tryLegacy => 1);
+    if (!$fnretpass) {
+        main_exit OVH::Bastion::EXIT_PASSFILE_NOT_FOUND, "passfile-not-found", $fnretpass->msg;
+    }
+    $passwordFile = $fnretpass->value;
+
+    # check if a specific password is forced
+    foreach my $grant (@{$fnret->value}) {
+
+        # only keep the grant matching the password clue and context AND with a forced password
+        if (
+            $grant->{'forcePassword'}
+            && (   ($userPasswordContext eq 'self' && $grant->{'type'} eq 'personal')
+                || ($userPasswordContext eq 'group' && $grant->{'type'} eq 'group-member' && $grant->{'group'} eq $userPasswordClue))
```

Comment:
A user can also be a group guest (i.e. have access to the group credentials, such as egress keys and passwords, but not automatically be granted the whole servers/device list, just a subset of those), in which case we should also enter this if.
It can be checked with `$grant->{'type'} eq 'group-guest'`

Reply:
While fixing this I noticed the guest grant does not contain the extra infos (the forced password)
To fix this, when fetching the guest grant, I replace it with the matching group grant

Reply:
Hah, actually you can't do that, it'll have adverse effects, this is because a guest access can carry more information that the member access, for a given server, such as a TTL expiration. e.g. you can have an ACL in the group like this:
127.0.0.0/8 port-any user-any
And a guest group access for some account with:
127.1.2.3 port 22 user root TTL=1d

Access will be allowed for the guest (as long as the TTL has not expired) because this guest ACL matches at least one of the regular group ACL. This also means that if the 127.0.0.0/8 ACL is removed from the group, the guest account will no longer have access to 127.1.2.3 through it, because even if this ACL is declared for their guest access, it no longer matches any ACL of the group itself.

In other words, guest group accesses may be more specific that the group ACL.

What this means for the force-password feature is that this should also be declared (if needed) when declaring the group guest access. Side note: the force-key feature has the same issue, as you can't use `--force-key` with `groupAddGuestAccess`, there is no way to force a key in guest accesses. We should fix that by adding the `--force-key` and `--force-password` options to `group{Add,Del}GuestAccess`. As this PR has already a lot of changes, I suggest doing it in a later one to avoid working on this one for 6 months :)

So in the meantime, so we can go forward, can you revert your change to `is_access_granted` by restoring groupGuest and groupLegacy vars instead of groupMember?

Note to self: I'll also add a test case that ensure such code change can no longer pass all the tests!